agent/internal/client/host_methods.go

```diff
@@ -292,23 +292,26 @@ func (c *hostCommunicator) Heartbeat(ctx context.Context, taskData TaskData) (bo
 	resp, err := c.request(ctx, info, data)
 	if err != nil {
 		err = errors.Wrapf(err, "error sending heartbeat for task %s", taskData.ID)
-		return false, err
+		return "", err
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode == http.StatusConflict {
-		return false, errors.Errorf("unauthorized - wrong secret")
+		return evergreen.TaskFailed, errors.Errorf("Unauthorized - wrong secret")
 	}
 	if resp.StatusCode != http.StatusOK {
-		return false, errors.Errorf("unexpected status code doing heartbeat: %v",
+		return "", errors.Errorf("unexpected status code doing heartbeat: %v",
 			resp.StatusCode)
 	}
 
 	heartbeatResponse := &apimodels.HeartbeatResponse{}
 	if err = utility.ReadJSON(resp.Body, heartbeatResponse); err != nil {
 		err = errors.Wrapf(err, "Error unmarshaling heartbeat response for task %s", taskData.ID)
-		return false, err
+		return "", err
 	}
-	return heartbeatResponse.Abort, nil
+	if heartbeatResponse.Abort {
+		return evergreen.TaskFailed, nil
```

Comment:
Is there a unit test for abort/conflict?

Reply:
Created